base_ami failing due to `oc ex dockerbuild` image name regex failures

https://ci.openshift.redhat.com/jenkins/job/base_ami/1990/console
```
Stdout from the command:

Building base images...
/data/src/github.com/openshift/origin ~
++ Building go targets for linux/amd64: cmd/oc
++ Placing binaries
/data/src/github.com/openshift/origin/hack/build-go.sh took 124 seconds


Stderr from the command:

--> Image centos:centos7 was not found, pulling ...
error: API error (500): repository name component must match "[a-z0-9](?:-*[a-z0-9])*(?:[._][a-z0-9](?:-*[a-z0-9])*)*"
!!! Error in hack/build-base-images.sh:28
	'"${oc}" ex dockerbuild $2 $1' exited with status 1
Call stack:
	1: hack/build-base-images.sh:28 build(...)
	2: hack/build-base-images.sh:32 main(...)
Exiting with status 1
Build step 'Execute shell' marked build as failure
```

@smarterclayton I'm seeing this in the log:
```
Building base images...
/data/src/github.com/openshift/origin ~
--> Image centos:centos7 was not found, pulling ...
error: API error (500): repository name component must match "[a-z0-9](?:-*[a-z0-9])*(?:[._][a-z0-9](?:-*[a-z0-9])*)*"
[ERROR] PID 2624: hack/build-base-images.sh:27: `'/data/src/github.com/openshift/origin/_output/local/bin/linux/amd64/oc' ex dockerbuild /data/src/github.com/openshift/origin/images/base openshift/origin-base` exited with status 1.
[INFO] 		Stack Trace: 
[INFO] 		  1: hack/build-base-images.sh:27: `'/data/src/github.com/openshift/origin/_output/local/bin/linux/amd64/oc' ex dockerbuild /data/src/github.com/openshift/origin/images/base openshift/origin-base`
[INFO] 		  2: hack/build-base-images.sh:31: build
[INFO]   Exiting with code 1.
```
 Why is that a different issue than this one? The failure cause for this one picks it up.

--- a/pkg/util/docker/dockerfile/builder/client.go
+++ b/pkg/util/docker/dockerfile/builder/client.go
@@ -117,7 +117,7 @@ func (e *ClientExecutor) Build(r io.Reader, args map[string]string) error {
 			}
 			from, err = e.CreateScratchImage()
 			if err != nil {
-				return err
+				return fmt.Errorf("unable to create a scratch image for this build: %v", err)
 			}
 			defer e.CleanupImage(from)
 		}
@@ -157,7 +157,7 @@ func (e *ClientExecutor) Build(r io.Reader, args map[string]string) error {
 				}
 				v, err := e.Client.CreateVolume(docker.CreateVolumeOptions{Name: volumeName})
 				if err != nil {
-					return err
+					return fmt.Errorf("unable to create volume to mount secrets: %v", err)
 				}
 				defer e.cleanupVolume(volumeName)
 				sharedMount = v.Mountpoint
@@ -181,7 +181,7 @@ func (e *ClientExecutor) Build(r io.Reader, args map[string]string) error {
 		}
 		container, err := e.Client.CreateContainer(opts)
 		if err != nil {
-			return err
+			return fmt.Errorf("unable to create build container: %v", err)
 		}
 		e.Container = container
 
@@ -200,7 +200,7 @@ func (e *ClientExecutor) Build(r io.Reader, args map[string]string) error {
 			})
 		}
 		if err := e.Copy(copies...); err != nil {
-			return err
+			return fmt.Errorf("unable to copy build context into container: %v", err)
 		}
 	}
 
@@ -223,7 +223,7 @@ func (e *ClientExecutor) Build(r io.Reader, args map[string]string) error {
 		}
 
 		if err := e.Client.StartContainer(e.Container.ID, &hostConfig); err != nil {
-			return err
+			return fmt.Errorf("unable to start build container: %v", err)
 		}
 		// TODO: is this racy? may have to loop wait in the actual run step
 	}
@@ -245,7 +245,7 @@ func (e *ClientExecutor) Build(r io.Reader, args map[string]string) error {
 	if mustStart {
 		glog.V(4).Infof("Stopping container %s ...", e.Container.ID)
 		if err := e.Client.StopContainer(e.Container.ID, 0); err != nil {
-			return err
+			return fmt.Errorf("unable to stop build container: %v", err)
 		}
 	}
 
@@ -272,7 +272,7 @@ func (e *ClientExecutor) Build(r io.Reader, args map[string]string) error {
 		Tag:        tag,
 	})
 	if err != nil {
-		return err
+		return fmt.Errorf("unable to commit build container: %v", err)
 	}
 	e.Image = image
 	glog.V(4).Infof("Committed %s to %s", e.Container.ID, e.Image.ID)
@@ -293,7 +293,7 @@ func (e *ClientExecutor) Cleanup() error {
 		Force:         true,
 	})
 	if _, ok := err.(*docker.NoSuchContainer); err != nil && !ok {
-		return err
+		return fmt.Errorf("unable to cleanup build container: %v", err)
 	}
 	e.Container = nil
 	return nil
@@ -389,7 +389,7 @@ func (e *ClientExecutor) LoadImage(from string) (*docker.Image, error) {
 	for _, config := range auth {
 		// TODO: handle IDs?
 		pullImageOptions := docker.PullImageOptions{
-			Repository:    from,
+			Repository:    repository,
 			Tag:           tag,
 			OutputStream:  imageprogress.NewPullWriter(outputProgress),
 			RawJSONStream: true,
@@ -406,7 +406,7 @@ func (e *ClientExecutor) LoadImage(from string) (*docker.Image, error) {
 		continue
 	}
 	if lastErr != nil {
-		return nil, lastErr
+		return nil, fmt.Errorf("unable to pull image (from: %s, tag: %s): %v", repository, tag, lastErr)
 	}
 
 	return e.Client.InspectImage(from)

